content/blog/2019-01-07-we-mastered-the-art-of-programming.md

```diff
@@ -0,0 +1,63 @@
+
+## problem solving at it's finest.
+
+What is connecting the dot between all this, is the curiosity and the satisfaction of solving a problem, a computer problem. We make things work. This is how we get points on a school assignment. This is how our boss sees us as the achiever. This is how we create pure value. Based on this, it only makes sense that the faster you solve a computer problem, the better you are.
+
+But computer problem seldom get stashed away once solved. They come back to haunt you and require adjustments to the code so it fits the new constraints. Great a new problem! You have to find the fastest way to hack the code. This is fun until one or two things happen: the code you have to modify comes from a *bad* programmer or something breaks every time you modify something else. It will eventually work, but it’ll take a little longer than the previous iteration. This continues, you realise that the *bad* programmer is your past self from a few month ago and you fear modifying anything since you have no faith this [*Jenga*](https://secure.img1-fg.wfcdn.com/im/93997415/resize-h800%5Ecompr-r85/4885/48852016/Jenga%25AE+giant%25u2122+premium+jeu+de+bois+franc.jpg) will hold at all.
+
+Maybe if you took a little more time to think of how future you would feel using it things would have been different. Maybe you would have cleaned the code. Maybe you would have explicated the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) (why divide by 8.2?). Maybe some side effects would have been commented. Well this is not important for now, because the code *works*.
+
+But this did not apply to me. I could jump back in my code, read the witty comment I wrote and remember what dimension of the matrix `X` were represented by variable `i`, `j` and `k`. I make functions, so I [don't repeat myself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). I had a bunch of smell tests that let me know if I break something. No really, I mastered the art of programming.
+
+## Until one day.
+
+I was working in a [Tech Startup](https://www.xpertsea.com/) and my team was working on a shrimp detection model. We were a few guys with different backgrounds. Everyone were software engineers except for myself; I was an actuary holding the most statistical knowledge. We were pitted against one another to figure how to solve this shrimp detection problem. Plot twist. I was not the one achieving — and for reasons I was not even aware of.
```

Comment:
Everyone was a software engineer

Reply:
Ex-actuary ? :P

Reply:
figure how : bizzare. solve the problem, figure out how

Reply:
I was not the one achieving : ça veut dire quoi ? I was not in the lead ?

Reply:
xie xie fixed